File: FinishedProject/CF Finder/Views/MapViewAnnotations/CFPinAnnotationView.m
//
//  Coffeeshop Finder
//
//  Created by Duc Nguyen on 9/5/16.
//  Copyright © 2016 Duc Nguyen. All rights reserved.
//
#import "CFPinAnnotationView.h"

#import "Constants.h"
#import "CFMapAnnotation.h"

@interface CFPinAnnotationView ()
{
    UILabel *minWalkLabel;
}

@end


@implementation CFPinAnnotationView

@synthesize leftCalloutAccessoryView = _leftCalloutAccessoryView;


-(UIView *)leftCalloutAccessoryView
{
    if (!_leftCalloutAccessoryView) {
        _leftCalloutAccessoryView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 40, self.frame.size.height + 11)];
        [_leftCalloutAccessoryView setBackgroundColor:UIColorFromRGB(RGB_RED)];
        
        UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"walkIcon"]];
        [imageView setCenter:CGPointMake(_leftCalloutAccessoryView.center.x, _leftCalloutAccessoryView.center.y - 5)];
        [_leftCalloutAccessoryView addSubview:imageView];
        
        minWalkLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, _leftCalloutAccessoryView.frame.size.height - 15, 40, 10)];
        [minWalkLabel setFont:[UIFont fontWithName:FONT_HELVETICA_MEDIUM size:8]];
        [minWalkLabel setTextColor:[UIColor whiteColor]];
        [minWalkLabel setTextAlignment:NSTextAlignmentCenter];
        [_leftCalloutAccessoryView addSubview:minWalkLabel];
    }
    [self updateMinWalkLabelText];
    return _leftCalloutAccessoryView;
}

-(void)updateMinWalkLabelText
{
    CFMapAnnotation *annotation = self.annotation;
    [minWalkLabel setText:[NSString stringWithFormat:@"%@ mts",annotation.distance]];
}

@end

File: FinishedProject/CF Finder/Views/CFPinAnnotationView.swift
//
//  CFPinAnnotationView.swift
//  CF Finder
//
//  Created by Duc Nguyen on 9/7/16.
//  Copyright © 2016 Duc Nguyen. All rights reserved.
//

import Foundation
import MapKit
import UIKit

class CFPinAnnotationView: MKPinAnnotationView {
    var minWalkLabel: UILabel!
    
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        
    }
    
    override init(annotation: MKAnnotation!, reuseIdentifier: String!)
    {
        super.init(annotation: annotation, reuseIdentifier: reuseIdentifier)
        
        let myleftCalloutAccessoryView = UIView(frame: CGRect(x: 0, y: 0, width: 40, height: self.frame.size.height + 11))
        myleftCalloutAccessoryView.backgroundColor = UIColor.redColor()
        
        myleftCalloutAccessoryView.backgroundColor = Constants.sharedInstance.UIColorFromRGB(Constants.sharedInstance.RGB_RED)
        let imageView = UIImageView(image: UIImage(named: "walkIcon")!)
        imageView.center = CGPoint(x: myleftCalloutAccessoryView.center.x, y: myleftCalloutAccessoryView.center.y - 5)
        myleftCalloutAccessoryView.addSubview(imageView)
        minWalkLabel = UILabel(frame: CGRect(x: 0, y: myleftCalloutAccessoryView.frame.size.height - 15, width: 40, height: 10))
        minWalkLabel.font = UIFont(name: Constants.sharedInstance.FONT_HELVETICA_MEDIUM, size: 8)
        minWalkLabel.textColor = UIColor.whiteColor()
        minWalkLabel.textAlignment =  .Center
        myleftCalloutAccessoryView.addSubview(minWalkLabel)
        self.updateMinWalkLabelText()
        
        
        self.leftCalloutAccessoryView = myleftCalloutAccessoryView
    }
    
    required init(coder aDecoder: NSCoder)
    {
        super.init(coder: aDecoder)!
    }
    
    
    func updateMinWalkLabelText() {
        let annotation: CFMapAnnotation = self.annotation as! CFMapAnnotation
        minWalkLabel.text = "\(annotation.distance!) mts"
    }
    
    
}